code/game/objects/buckling.dm

```diff
@@ -138,18 +133,75 @@
 //same but for unbuckle
 /atom/movable/proc/post_unbuckle_mob(mob/living/M)
 
+/**
+  *	Simple helper proc that ruins a suite of checks to test whether it is possible or not to buckle the target mob to src.
+  *
+  * Returns FALSE if any conditions that should prevent buckling are satisfied. Returns TRUE otherwise.
+  * Arguments:
+  * * target - Target mob to check against buckling to src.
+  * * force - Whether or not the buckle should be forced. If TRUE, ignores src's can_buckle var.
+  * * check_loc - Whether to do a proximity check or not. The proximity check looks for target.loc == src.loc.
+  */
+/atom/movable/proc/is_buckle_possible(mob/living/target, force = FALSE, check_loc = TRUE)
+	if(!istype(target))
+		return FALSE
+
+	if(check_loc && target.loc != loc)
+		return FALSE
+
+	if((!can_buckle && !force) || target.buckled || (LAZYLEN(buckled_mobs) >= max_buckled_mobs) || (buckle_requires_restraints && !target.restrained()) || target == src)
```

Comment:
Can you split this into multiple conditions? This one is pretty bulky.

Reply:
I've fully split it all out and added a bit of commenting as to what everything does.

Also identified the issue with aggro grab buckling to pipes earlier today, so I included that single word fix too since I had to push a new commit and I was reorganising that code anyway.